packages/scene-composer/src/components/panels/scene-settings/MatterportIntegration.tsx

```diff
@@ -0,0 +1,220 @@
+
+  const [dirty, setDirty] = useState(false);
+  const [matterportModelId, setMatterportModelId] = useState(getSceneProperty(KnownSceneProperty.MatterportModelId));
+  const [connectionOptions, setConnectionOptions] = useState<{ label: string; value: string }[]>([
+    {
+      label: intl.formatMessage({
+        defaultMessage: 'Select a connection',
+        description: 'Select a connection placeholder',
+      }),
+      value: 'n/a',
+    },
+  ]);
+  const [selectedConnectionName, setSelectedConnectionName] = useState('n/a');
+
+  const updateSelectedConnectionName = async () => {
+    const getSceneInfoFunction = getGlobalSettings().getSceneInfoFunction;
+    if (getSceneInfoFunction) {
+      const getSceneresponse = await getSceneInfoFunction();
+      if (getSceneresponse && getSceneresponse.sceneMetadata && getSceneresponse.sceneMetadata.MATTERPORT_SECRET_ARN) {
+        setSelectedConnectionName(getSceneresponse.sceneMetadata.MATTERPORT_SECRET_ARN);
+      } else {
+        setSelectedConnectionName('n/a');
+      }
+    }
+  };
+
+  const getConnectionList = async () => {
+    const connectionList: { label: string; value: string }[] = [
+      {
+        label: intl.formatMessage({
+          defaultMessage: 'Select a connection',
+          description: 'Select a connection placeholder',
+        }),
+        value: 'n/a',
+      },
+    ];
+    const get3pConnectionListFunction = getGlobalSettings().get3pConnectionListFunction;
+    if (get3pConnectionListFunction) {
+      const response = await get3pConnectionListFunction('AWSIoTTwinMaker_Matterport');
+      if (response) {
+        response.forEach((secret) => {
+          if (secret.Name && secret.ARN) {
+            connectionList.push({ label: secret.Name, value: secret.ARN });
+          }
+        });
+      }
+    }
+    setConnectionOptions(connectionList);
+  };
+
+  const onUpdated = () => {
+    updateSelectedConnectionName();
+    getConnectionList();
+  };
+
+  useEffect(() => {
+    subscribe(onUpdated);
+    updateSelectedConnectionName();
+    getConnectionList();
+
+    return () => unsubscribe(onUpdated);
+  }, []);
+
+  const onConnectionNameChange = useCallback(async (event) => {
+    const value = event.detail.selectedOption.value;
+    setSelectedConnectionName(value);
+
+    const getSceneInfoFunction = getGlobalSettings().getSceneInfoFunction;
+    const updateSceneFunction = getGlobalSettings().updateSceneFunction;
+    if (getSceneInfoFunction && updateSceneFunction) {
+      let sceneCapabilities: string[] | undefined;
+      let sceneMetadata: Record<string, string> | undefined;
+      const getSceneresponse = await getSceneInfoFunction();
+      if (getSceneresponse) {
+        sceneCapabilities = getSceneresponse.capabilities;
+        sceneMetadata = getSceneresponse.sceneMetadata;
+      }
+
+      if (value === 'n/a') {
+        if (sceneCapabilities && sceneCapabilities.includes('MATTERPORT')) {
```

Comment:
constant for `MATTERPORT`

Reply:
Done - Moved to constants
`export const SCENE_CAPABILITY_MATTERPORT = 'MATTERPORT';`